indico/migrations/versions/20230306_0905_e47fc6634291_add_menu_entry_acls.py

```diff
@@ -0,0 +1,352 @@
+    conn.execute(
+        sa.text(
+            '''
+                    UPDATE menu_entries
+                       SET speaker_allowed = TRUE
+                     WHERE access = :speakers
+                       AND id NOT IN (
+                         SELECT menu_entry_id AS id
+                           FROM menu_entries_principals mep
+                          WHERE mep.type = :regform
+                       )
+            '''
+        ),
+        speakers=_MenuEntryAccess.speakers.value,
+        regform=_PrincipalType.registration_form.value,
+    )
+
+
+def _downgrade_access_data():
+    """
+    Downgrade the protection_mode, speaker_allowed and principals to the access column. This isn't
+    lossless given the new feature is more fine-grained, but best effort.
+
+    Here is the mapping:
+
+         protection_mode | speaker_allowed | menu_entry_principals     || access
+        =================+=================+===========================++========================
+          protected      | true            | doesn't contain regform   ||  speakers
+                         |                 | principals                ||
+        -----------------+-----------------+---------------------------++------------------------
+          inheriting     |                 | doesn't contain regform   ||  speakers
+        parent=protected | parent=true     | principals                ||
+        -----------------+-----------------+---------------------------++------------------------
+          protected      | regardless      | at least one regform      ||  registered_participants
+                         |                 | principal                 ||
+        -----------------+-----------------+---------------------------++------------------------
+          inheriting     | regardless      | parent has at least one   ||  registered_participants
+                         |                 | regform principal         ||
+        -----------------+-----------------+---------------------------++------------------------
+          protected      | false           | doesn't contain regform   ||  everyone
+                         |                 | principals                ||  (no good mapping)
+        -----------------+-----------------+---------------------------++------------------------
+          inheriting     | false           | doesn't contain regform   ||  everyone
+                         |                 | principals                ||
+        -----------------+-----------------+---------------------------++------------------------
+          all other combinations                                       ||  everyone
+                                                                       ||  (no good mapping)
+        -----------------+-----------------+---------------------------++------------------------
+    """
+    conn = op.get_bind()
+    conn.execute('SET search_path TO events')
```

Comment:
I'd prefer full `schema.table` names :)

Reply:
As do I, though there was some SQL error I couldn't get past. I tried to reproduce the error now but am no longer getting it, yay! Updated the patch.